### DISCUSSION TITLE
twinBASIC BETA 67

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#959] [Releases]
- added: Shape control (currently not windowless) [ https://github.com/WaynePhillipsEA/twinbasic/issues/938 ]
- fixed: several ActiveX tab-key related issues [ https://github.com/WaynePhillipsEA/twinbasic/issues/935 ]
- fixed: form designer duplicate/paste issue with the naming of the new controls [ https://github.com/WaynePhillipsEA/twinbasic/issues/957 ]

<hr /><em>This discussion was created from the release <a href='https://github.com/WaynePhillipsEA/twinbasic/releases/tag/beta-67'>twinBASIC BETA 67</a>.</em>

**Comment by WaynePhillipsEA** [comment]
There is a bug with the Shape control in this release, in that the Circle/Square/RoundedSquare shapes are not centered correctly.  Fixed in next release.

**Comment by FullValueRider** [comment]
Microsoft Defender is resolutely refusing to allow a download of Beta 67.  This is the first time I've encountered any antivirus stuff with twinBasic downloads

Windows 11
![Beta67](https://user-images.githubusercontent.com/81257042/176679605-f4ed1bb7-a90b-425c-a7e5-d9b179b17893.jpg)

**Reply by bclothier** [reply]
Not sure that's Windows Defender's doing. It looks like it's Google Chrome that's giving you the hard time. Windows Defender wouldn't send it to Google to scan for viruses. Normally there's a "keep anyway" button but I don't see it. Is it available via the ^ button?

**Reply by VB6-Programming** [reply]
Downloaded OK for me - Microsoft Edge, Microsoft Defender, Windows 10.

Got usual message from Defender on first run of twinBASIC new beta - and chose Run Anyway.

**Reply by Mo-Lo** [reply]
![image](https://user-images.githubusercontent.com/83885514/176758342-a175bce0-236e-4c88-bbcb-0412b954bd01.png)

**Reply by FullValueRider** [reply]
Yes,  I've used that option in other situations.  In this case there were no options offered to unquarantine the file or the option to unquarantine was greyed out..  That's why the original post included a view of multiple attempts at downloading.  If it hadn't been such an unusual event I wouldn't have posted it as an issue.

Attempting to download beta67 in chrome today (24 hours or thereabouts after the above) resulted in a straightforward and successful download.

**Comment by FullValueRider** [comment]
Windows Defender does pop up
![Beta67defender](https://user-images.githubusercontent.com/81257042/176685186-6dd46f11-93e1-47ff-be0e-706632cb5e98.jpg)

**Reply by WaynePhillipsEA** [reply]
ML at the end of the virus code stands for Machine Learning.  Basically, it's a guess, rather than being a confirmed virus.  I wish the anti virus vendors would make it clearer, rather than scaremongering.

**Reply by bclothier** [reply]
They don't and won't trust us plebs to actually read and understand the heady concepts of a trojan horse, machine learning, and all other super-complicated stuff that only Comp. Sci. Ph.Ds can comprehend  so scare us they must. Else, it would expose the scam of their whole business model which consist of them standing around with a finger in wind.

**Comment by FullValueRider** [comment]
Installed Firefox and Beta67 downloaded and installed without issue.

**Comment by mburns08109** [comment]
Wayne,

In terms of AV "missing pieces" things are looking MUCH better in 66 AND 67!
The error messages are more accurate, and the exit picture much more reasonable.

I'll count this as "fixed for now, at least".   :-)

**Reply by WaynePhillipsEA** [reply]
Great, thanks for the update!

**Comment by WaynePhillipsEA** [comment]
With regards to the anti-virus false positives, I'm wondering whether these new message boxes included in the latest releases could be the cause of red flags:

`ERROR file xyz is missing.  Check that the file exists and hasn't been quarantined by your anti-virus software`.

Perhaps just the mention of virus or quarantine is not helping?... WTFDIK.

**Reply by Mo-Lo** [reply]
From my perspective, the new message boxes have nothing to do with that. AV is just stubborn and quarantines the file after every single download! So, I just ignore the blah blah and restore the file.

**Reply by EduardoVB** [reply]
> `ERROR file xyz is missing. Check that the file exists and hasn't been quarantined by your anti-virus software`.
> 
> Perhaps just the mention of virus or quarantine is not helping

IMO it is a possibility.
You could obfuscate the message and compose/decode it just before showing it. Specially the part "quarantined by your anti-virus".

**Reply by WaynePhillipsEA** [reply]
> You could obfuscate the message and compose/decode it just before showing it.

That may look even more suspicious! (assuming the modern AV engines do some basic simulation of code).  

Perhaps we can just say `ERROR missing file xyz... Visit twinBASIC.com/turn_off_your_stupid_antivirus.htm for more information`

**Reply by EduardoVB** [reply]
> That may look even more suspicious! (assuming the modern AV engines do some basic simulation of code).

Maybe, who knows. Or it could be a fix for some AVs but raise alarms in others.
The link maybe a better idea, but instead of turn_off_your_stupid_antivirus.htm better something like about_security.htm 🤔😁

**Reply by bclothier** [reply]
And they might flag a link as potentially dangerous. Either avoid using pretty link (e.g. `<a href="www.nothere.com">Go to here.com to learn more</a>`) or render the URL as plain text. 

As was mentioned somewhere else already, it might be prudent to start signing the executables and building reputation. Maybe automate sending the files to their respective sites to let them know it's OK.

**Reply by EduardoVB** [reply]
Signing is supposed to help but does not guarantee that it won't be ever flagged.

The prices start at $70/year.
https://cheapsslsecurity.com/sectigo/sectigo-code-signing-certificate.html

**Comment by mansellan** [comment]
Heh, I just had to start signing executables at work to get ClickOnce to not be so alarmist. There _used_ to be a difference between standard and EV certs, not sure if that makes a difference to AV though. EV is more expensive and more painful to acquire and set up.

We went with Digicert:  https://www.digicert.com/signing/code-signing-certificates

Not the cheapest but they are our preferred vendor for TLS so it was easier for my company to stick with them

**Reply by bclothier** [reply]
Don't think EV applies to code signing certificates?

**Reply by mansellan** [reply]
You can get an EV code cert, but I'm not sure if anything looks for it any more

**Comment by bclothier** [comment]
> Signing is supposed to help but does not guarantee that it won't be ever flagged.

Correct. It won't make it go away but I understand that once you have a bit of reputation, it makes it much less likely that your product will get hard-blocked by various AVs like what apparently happened. Ultimately, it's still a game of musical chairs.

**Reply by mburns08109** [reply]
On that point, by my observations of recent BETAs, the code-signing has ended all objections to TB executables in the beta distributions by Norton360, at least (so far, anyway).

**Reply by WaynePhillipsEA** [reply]
@mburns08109 that is good news, thanks.
